### PR TITLE
Сhanging the switch logic for BuildFilePlatformFilter

### PR DESCRIPTION
**Pull request by dever25**
Resolves https://github.com/tuist/tuist/issues/4708

### Short description 📝

Сhanging the switch logic

### How to test the changes locally 🧐

When creating a "Deployment Target.os(targetVersion: "14.0", devices: [.iphone, .mac, .iPad]" the filter parameter is set to "always used".

### CODE REVIEW

File: Sources/TuistGraph/Models/Target.swift [hunk 1]
```diff
@@ -184,6 +184,8 @@ public struct Target: Equatable, Hashable, Comparable, Codable {
     /// with Catalyst compatibility.
     public var targetDependencyBuildFilesPlatformFilter: BuildFilePlatformFilter? {
         switch deploymentTarget {
+        case let .iOS(_, devices) where devices.contains(.all):
+            return nil
```
Comment (kwridan) [inline Sources/TuistGraph/Models/Target.swift:188]: Would the following case also require the same treatment?

```swift
.iOS(targetVersion: "15.5", devices: [.ipad, .mac]), 
```

The bit I'm struggling with is understanding the scenarios where `.catalyst` filter is required? 

Looking at the history,
https://github.com/tuist/tuist/commit/5220cca4ec8a292978744a2ef0804c9d4a159f78 it seems this was added to achieve the inverse, to exclude certain dependencies from catalyst. Which is making me think is the logic below perhaps incorrect?

By any chance are you able to create a small fixture where the different use cases are applicable to help us better test this?

**Review by kwridan — COMMENTED** [review]
Thanks @dever25

File: Sources/TuistGraph/Models/Target.swift (hunk 1 above)
Comment (dever25) [reply Sources/TuistGraph/Models/Target.swift:188]: If we return nil, the "always used" parameter is set
If we return the catalyst, the parameter "Mac catalyst" is set
If we return ios, the "iOS" parameter is set

When I write [.iphone, .mac, .iPad], I want the "Always used" parameter to be set. To do this, you need to return nil.
It also seems to me that "nil" should be returned for combinations of [.ipad, .Mac] and [.iPhone, .Mac]. But I did not prescribe it, so as not to break the logic of other users

How do I see the parameter dependencies:
[.iphone, .mac, .iPad] -> iOS
[.iphone, .mac, .iPad] -> iOS
[.iphone, .mac, .iPad] -> iOS
[.iphone, .mac, .iPad] -> iOS

Comment (danyf90) [reply Sources/TuistGraph/Models/Target.swift:188]: I think @kwridan point is that with your changes `[ipad, mac]`, or `[iphone, mac]` would return `catalyst`.

The first contdition shouldn't check if they are all there, but if there is both catalyst and at least one between iphone and ipad.

Makes sense?

Comment (dever25) [reply Sources/TuistGraph/Models/Target.swift:188]: Снимок экрана 2022-09-02 в 16 34 01

@danyf90 It seems that [ipad, mac] should return catalyst
And it is impossible to choose [iphone, mac] without iPad(see picture)

**Review by danyf90 — APPROVED** [review]

**Pull request merged**